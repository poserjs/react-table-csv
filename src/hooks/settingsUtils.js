export const SETTINGS_VERSION = '0.1';

export const buildSettings = ({
  currentTheme,
  columnStyles,
  columnOrder,
  hiddenColumns,
  filters,
  dropdownFilters,
  filterMode,
  showFilterRow,
  pinnedAnchor,
  showRowNumbers,
  customize,
  tableMaxHeight,
  tableMaxWidth,
  fontSize,
}) => {
  const dropdown = {};
  if (dropdownFilters && typeof dropdownFilters === 'object') {
    Object.entries(dropdownFilters).forEach(([k, v]) => {
      dropdown[k] = Array.from(v || []);
    });
  }
  return {
    version: SETTINGS_VERSION,
    theme: currentTheme,
    columnStyles,
    columnOrder,
    hiddenColumns: Array.from(hiddenColumns || []),
    filters,
    dropdownFilters: dropdown,
    filterMode,
    showFilterRow,
    pinnedAnchor,
    showRowNumbers,
    customize,
    tableMaxHeight,
    tableMaxWidth,
    fontSize,
  };
};

