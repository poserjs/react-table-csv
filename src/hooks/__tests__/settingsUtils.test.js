/* eslint-env jest */
import { buildSettings } from '../settingsUtils';

describe('buildSettings', () => {
  it('builds settings from state', () => {
    const state = {
      currentTheme: 'dark',
      columnStyles: { col1: { width: 100 } },
      columnOrder: ['col1', 'col2'],
      hiddenColumns: new Set(['col2']),
      filters: { col1: 'foo' },
      dropdownFilters: { col2: new Set(['a', 'b']) },
      filterMode: { col1: 'contains' },
      showFilterRow: true,
      pinnedAnchor: 'col1',
      showRowNumbers: true,
      customize: true,
    };
    expect(buildSettings(state)).toMatchSnapshot();
  });

  it('converts dropdown filters to arrays', () => {
    const state = {
      currentTheme: 'lite',
      columnStyles: {},
      columnOrder: [],
      hiddenColumns: new Set(),
      filters: {},
      dropdownFilters: { col1: new Set(['x']) },
      filterMode: {},
      showFilterRow: false,
      pinnedAnchor: null,
      showRowNumbers: false,
      customize: false,
    };
    const result = buildSettings(state);
    expect(Array.isArray(result.dropdownFilters.col1)).toBe(true);
    expect(result.dropdownFilters.col1).toEqual(['x']);
  });
});

