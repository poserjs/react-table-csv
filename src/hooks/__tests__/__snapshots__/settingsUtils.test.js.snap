// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`buildSettings builds settings from state 1`] = `
{
  "columnOrder": [
    "col1",
    "col2",
  ],
  "columnStyles": {
    "col1": {
      "width": 100,
    },
  },
  "customize": true,
  "dropdownFilters": {
    "col2": [
      "a",
      "b",
    ],
  },
  "filterMode": {
    "col1": "contains",
  },
  "filters": {
    "col1": "foo",
  },
  "fontSize": 14,
  "hiddenColumns": [
    "col2",
  ],
  "pinnedAnchor": "col1",
  "showFilterRow": true,
  "showRowNumbers": true,
  "tableMaxHeight": "500px",
  "tableMaxWidth": "75%",
  "theme": "dark",
  "version": "0.1",
}
`;
